hax/helper/generate_sysconf.py

```diff
@@ -0,0 +1,255 @@
+        profile_key = self.kv.kv_get('m0conf/profiles/', keys=True)
+        assert profile_key
+        # Get the first profile fid
+        profile_fid = profile_key[0].split('/')[-1]
+        fid = Fid(ObjT.PROCESS.value, int(svc_id))
+        ep = self.get_service_ep(svc_id)
+        filename = f's3server-{fid}'
+        contents = (f'MOTR_PROFILE_FID={profile_fid}\n'
+                    f'MOTR_S3SERVER_EP={ep}\n'
+                    f'MOTR_HA_EP={hax_ep}\n'
+                    f'MOTR_PROCESS_FID={fid}\n'
+                    f'MOTR_S3SERVER_PORT={s3_port}\n')
+        self._write_file(self.sysconfig_dir + filename, contents)
+
+    def generate_sysconfig(self):
+        s3_port = 28071
+        IDs = self.get_all_svc_ids()
+
+        hax_ep = self.get_service_ep(IDs['HAX_ID'][0])
+        for x in IDs['CONFD_IDs']:
+            self.generate_confd(x, hax_ep)
+
+        for x in IDs['IOS_IDs']:
+            self.generate_ios(x, hax_ep)
+
+        for x in IDs['S3_IDs']:
+            self.generate_s3(x, hax_ep, s3_port)
+            s3_port += 1
+
+    def append_svcs_to_file(self, svcs, conf_file):
+        with open(conf_file) as consul_conf_file:
+            data = simplejson.load(consul_conf_file)
+        data['services'] = svcs
+        self._write_file(conf_file,
+                         simplejson.dumps(data, indent=2, for_json=True))
+
+    def get_service_ipaddr(self, ep: str) -> str:
+        return ep.split('@')[0]
+
+    def append_ipaddr_to_file(self, hax_ep: str, conf_file: str):
+        ipaddr = self.get_service_ipaddr(hax_ep)
+        with fileinput.FileInput(conf_file, inplace=True,
+                                 backup='.bak') as file:
+            for line in file:
+                print(line.replace('http://localhost',
+                                   f'http://{ipaddr}'), end='')
+
+    def get_service_addr(self, ep: str) -> str:
+        return ep.rsplit('@', 1)[0]
+
+    def get_service_port(self, ep: str) -> int:
+        return int(ep.rsplit('@', 1)[1])
+
+    def prepare_svc(self, id: str, name: str):
+        ep = self.get_service_ep(id)
+        addr = self.get_service_addr(ep)
+        port = self.get_service_port(ep)
+
+        checks = {}
+        checks['args'] = ['/opt/seagate/cortx/hare/libexec/check-service']
+        checks['interval'] = '1s'
+        checks['status'] = 'warning'
+        # get svc checks args as per svc name
+        if name == 'hax':
+            checks['args'].append('--hax')
+        elif name in ('confd', 'ios'):
+            fid = Fid(ObjT.PROCESS.value, int(id))
+            checks['args'].extend(['--fid', fid])
+        elif name == 's3service':
+            fid = Fid(ObjT.PROCESS.value, int(id))
+            s3svc = 's3server@'+fid
+            checks['args'].extend(['--svc', s3svc])
+        return Service(id=id, name=name, address=addr, port=port,
+                       checks=[checks])
+
+    def update_consul_conf(self, conf_dir='/var/lib/hare'):
```

Comment:
Make conf_dir a mandatory parameter and don't set a default value. Caller must decide on passing the value.

Reply:
yes done